js-advanced-functions/fn-exercise-2.js

```diff
@@ -0,0 +1,62 @@
+console.log('CALL METHOD IMPLEMENTATION----------------------')
+const user={
+    firstName:'Aleksandar',
+    lastName:'Todorovski',
+    age:28,
+    print(){
+        console.log(`First name:${this.firstName} \nLast name:${this.lastName}`)
+    },
+    getUserData(field,fallbackMsg){
+        field=this?.firstName 
```

Comment:
this is fine and works as expected. Just small code improvements i think it would be good to mention so you know.
Instead of trying to get `this?.firstName` then `this?.lastName`, etc. You can dynamically access to object keys using variables. Check this link https://stackoverflow.com/questions/4244896/dynamically-access-object-property-using-variable

So here you would have only one line of code `this[field] ?? fallbackMsg` And that would be the easiest solution

Reply:
Thank you for the note, I tried it, but then the logic doesn't work at printing the data. This is the result:

Get first name: isLoggedIn field doesn't exist on user1
Get isLoggedIn: isLoggedIn field doesn't exist on user1

Reply:
Instead of doing `this.firstName` you can do `this[variable]` which means we can dynamically access property in an object. - this should be handled here in the `getUserData` method.
Then, when calling the method you need to statically pass the name of the field like `user.getUserData('firstName', 'Firstname field doesnt exist on user')`

Reply:
@atodoroski I think I got it. You can see the update in the last commit